packages/measured-reporting/lib/validators/inputValidators.js

```diff
@@ -0,0 +1,194 @@
+const Optional = require('optional-js');
+
+/**
+ * This module contains various validators to validate publicly exposed input
+ * @module inputValidators
+ */
+module.exports = {
+  /**
+   * Validates @{link Gauge} options.
+   *
+   * @param {string} name The metric name
+   * @param {function} callback The callback for the Gauge
+   * @param {Dimensions} dimensions The optional custom dimensions
+   * @param {number} publishingIntervalInSeconds the optional publishing interval
+   */
+  validateGaugeOptions: (name, callback, dimensions, publishingIntervalInSeconds) => {
+    module.exports.validateMetricName(name);
+    module.exports.validateNumberReturningCallback(callback);
+    module.exports.validateOptionalDimensions(dimensions);
+    module.exports.validateOptionalPublishingInterval(publishingIntervalInSeconds);
+  },
+
+  /**
+   * Validates the create histogram Options.
+   *
+   * @param {string} name The metric name
+   * @param {Dimensions} dimensions The optional custom dimensions
+   * @param {number} publishingIntervalInSeconds the optional publishing interval
+   */
+  validateHistogramOptions: (name, dimensions, publishingIntervalInSeconds) => {
+    module.exports.validateMetricName(name);
+    module.exports.validateOptionalDimensions(dimensions);
+    module.exports.validateOptionalPublishingInterval(publishingIntervalInSeconds);
+  },
+
+  /**
+   * Validates the create counter Options.
+   *
+   * @param {string} name The metric name
+   * @param {Dimensions} dimensions The optional custom dimensions
+   * @param {number} publishingIntervalInSeconds the optional publishing interval
+   */
+  validateCounterOptions: (name, dimensions, publishingIntervalInSeconds) => {
+    module.exports.validateMetricName(name);
+    module.exports.validateOptionalDimensions(dimensions);
+    module.exports.validateOptionalPublishingInterval(publishingIntervalInSeconds);
+  },
+
+  /**
+   * Validates the create timer Options.
+   *
+   * @param {string} name The metric name
+   * @param {Dimensions} dimensions The optional custom dimensions
+   * @param {number} publishingIntervalInSeconds the optional publishing interval
+   */
+  validateTimerOptions: (name, dimensions, publishingIntervalInSeconds) => {
```

Comment:
I might be missing something but it looks like these validate methods have duplicate logic.  Can they be collapsed into a single `validateMetricsOptions()` method that is shared by the different metric types.

Reply:
- [ ] collapsed and took care of todos